lua/telescope/pickers.lua

```diff
@@ -401,18 +408,53 @@ function Picker:find()
       end
 
       -- TODO: Entry manager should have a "bulk" setter. This can prevent a lot of redraws from display
-      self.manager = EntryManager:new(self.max_results, self.entry_adder, self.stats)
+      if self.cache_picker == false or not (self.cache_picker.is_cached == true) then
+        self.manager = EntryManager:new(self.max_results, self.entry_adder, self.stats)
 
-      local find_id = self:_next_find_id()
-      local process_result = self:get_result_processor(find_id, prompt, debounced_status)
-      local process_complete = self:get_result_completor(self.results_bufnr, find_id, prompt, status_updater)
+        local find_id = self:_next_find_id()
+        local process_result = self:get_result_processor(find_id, prompt, debounced_status)
+        local process_complete = self:get_result_completor(self.results_bufnr, find_id, prompt, status_updater)
 
-      local ok, msg = pcall(function()
-        self.finder(prompt, process_result, vim.schedule_wrap(process_complete))
-      end)
+        local ok, msg = pcall(function()
+          self.finder(prompt, process_result, vim.schedule_wrap(process_complete))
+        end)
 
-      if not ok then
-        log.warn("Failed with msg: ", msg)
+        if not ok then
+          log.warn("Failed with msg: ", msg)
+        end
+      else
```

Comment:
Hmmm, this section makes me a bit nervous.

We have a few vim.schedules, which I'm hoping we should not have to do necessarily.

Also, with newest update (you'll have to pull from master again) you should be able to do something like await_scheduler() instead, if you really need to do scheduling to make this work out OK.

Also, I'd see if we can re-use more code here or abstract it into some helper function within picker so that we don't share too much state here accidentally.

Does that make sense?

Reply:
> re-use more code here

The function for highlighting was already there :laughing: I just hadn't seen it.

Yes, `await_scheduler` is required to make this work.